### PR TITLE
nixos/intel-gpu-tools: init basic security wrapper

### LINKED ISSUE #309034: mangohud: Failed to initialize PMU! (Permission denied)
### Describe the bug

For Intel dGPUs, `mangohud` requires `CAP_PERFMON` in order to access GPU utilization.

### Steps To Reproduce

1. Install an Intel dGPU, such as the Arc ~~largest civilian airliner~~ A380
2. Install `mangohud` and `vkcube`
3. Run `mangohud vkcube`
4. See the following relevant logs:

```
Selected GPU 0: Intel(R) Arc(tm) A380 Graphics (DG2), type: DiscreteGpu
Failed to initialize PMU! (Permission denied)

When running as a normal user CAP_PERFMON is required to access performance
monitoring. See "man 7 capabilities", "man 8 setcap", or contact your
distribution vendor for assistance.

More information can be found at 'Perf events and tool security' document:
https://www.kernel.org/doc/html/latest/admin-guide/perf-security.html
```

### Expected behavior

This error should not happen, and `mangohud` should be able to read GPU utilization.

### Screenshots

Screenshot from 2024-05-04 07-46-43

As you can see, GPU utilization information is not being displayed.

### Additional context

*N.A.*

### Notify maintainers

CC @kira-bruneau @zeratax

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[gavin@gavin-gaming-desktop:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.30, NixOS, 24.05 (Uakari), 24.05.20240503.e96601e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/3gcgx970vsz0h89kjm3nwixb74b03g7r-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kira-bruneau**
## Description of changes

A simple NixOS module that adds a security wrapper to `intel_gpu_top` - to support collecting GPU stats from intel GPUs.

@pSub Some other binaries might also need to be wrapped here, but I was focused on just `intel_gpu_top` to resolve https://github.com/NixOS/nixpkgs/issues/309034.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/hardware/gpu/intel-gpu-tools.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
```
Comment (eclairevoyant) [inline nixos/modules/hardware/gpu/intel-gpu-tools.nix:3]:
```suggestion
```

top-level `with lib;` is discouraged, let's avoid this.

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 2]
```diff
@@ -155,6 +155,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - [Monado](https://monado.freedesktop.org/), an open source XR runtime. Available as [services.monado](#opt-services.monado.enable).
 
+- [intel-gpu-tools](https://drm.pages.freedesktop.org/igt-gpu-tools), tools for development and testing of the Intel DRM driver. Available as [hardware.intel-gpu-tools](#opt-hardware.intel-gpu-tools.enable)
```
Comment (eclairevoyant) [inline nixos/doc/manual/release-notes/rl-2405.section.md:158]: igt isn't really a service? but I'm not sure about how release notes are handled for non-service modules, just keeping this as a note.

File: nixos/modules/hardware/gpu/intel-gpu-tools.nix (hunk 1 above)
Comment (kira-bruneau) [reply nixos/modules/hardware/gpu/intel-gpu-tools.nix:3]: This is something I've always been pretty conflicted on. It's discouraged in general, but like the common use of `with lib` for `meta`, almost all (~80%) nixos modules follow this pattern and it seems more out of place to not do this.

I'm ok with changing this though!

File: nixos/doc/manual/release-notes/rl-2405.section.md (hunk 2 above)
Comment (kira-bruneau) [reply nixos/doc/manual/release-notes/rl-2405.section.md:158]: Yeah, I was a bit unsure about this too, but historically setcap wrappers like this have also been added to the new services section. It seems like this section has always been catch all for any new module whether or not it's actually a service.

Comment (kira-bruneau) [reply nixos/doc/manual/release-notes/rl-2405.section.md:158]: 20.09 is the only release that further breaks it down into subsections: https://github.com/NixOS/nixpkgs/blob/a0c8b880a41471e9754b31af84e8ea52132943c2/nixos/doc/manual/release-notes/rl-2009.section.md#new-services-sec-release-2009-new-services

**Review by Pandapip1 — APPROVED** [review]
I know I don't have any permissions here, but LGTM.

**Pull request merged**